soundcard/pulseaudio.py

```diff
@@ -367,11 +372,17 @@ def record(self, numframes):
                     chunk = numpy.zeros(nbytes_ptr[0]//4, dtype='float32')
                 if nbytes_ptr[0] > 0:
                     self._pulse._pa_stream_drop(self.stream)
-                    captured_data.append(chunk)
-                    captured_frames += len(chunk)/self.channels
+                    self.captured_data.append(chunk)
+                    self.captured_frames += len(chunk)/self.channels
             else:
                 time.sleep(0.001)
-        return numpy.reshape(numpy.concatenate(captured_data), [-1, self.channels])
+        self.last_record = time.time()
+        self.captured_data = numpy.concatenate(self.captured_data)
+        to_return, self.captured_data = numpy.split(self.captured_data,
+                                                    [int(numframes*self.channels)])
```

Comment:
I don't know the exact performance implications of this, but it might be preferable to split only the last entry in `self.captured_data`, and then `concatenate` all but the last entry.